src/handler.rs

```diff
@@ -327,6 +338,9 @@ impl RequestHandler {
             // Append CORS headers if they are present
             let resp = cors::post_process(&self.opts, req, resp)?;
 
+            // Set Content-Type for markdown files
+            let resp = crate::markdown::post_process(&self.opts, req, resp)?;
+
```

Comment:
(2) Once `markdown::post_process` function is updated ([comment 1](https://github.com/static-web-server/static-web-server/pull/577/files#r2495496659)), then the caller side here can be changed to:

```suggestion
            let resp = markdown::post_process(uri_path_md.is_some(), &self.opts, resp)?;

```